### PR TITLE
Logging improvements to vUSB

### PR DESCRIPTION
**Pull request by pearsonk**
- Add logging to specific desired events
-- Device added
-- Device removed
-- Device assigned to VM
-- Device unassigned from VM
-- Policy rule added
-- Policy rule removed
-- Policy match for decision to allow assignment
-- Policy match for decision to deny assignment
-- Implicit policy deny due to no matching policy rule
- Move some low-level logs to DEBUG priority
- Move some error logs to ERROR priority

Signed-off-by: Kevin Pearson <kevin.pearson@ortmanconsulting.com>

### CODE REVIEW

**Review by dpsmith — APPROVED** [review]
Reviewed-by Daniel P. Smith <dpsmith@apertussolutions.com>

**Comment by eric-ch** [comment]
Build [$10131](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10131)

File: src/policy.c [hunk 1]
```diff
@@ -522,6 +563,16 @@ policy_auto_assign_new_device(device_t *device)
...
+    xd_log(LOG_INFO,
+        "Automatically assigned device [Bus=%03d, Dev=%03d, VID=%04X, PID=%04X, Serial=%s] to VM [UUID=%s, DomID=%d], according to policy rule %d",
+        device->busid,
+        device->devid,
+        device->vendorid,
+        device->deviceid,
+        device->serial,
+        vm->uuid,
+        vm->domid,
+        rule->pos);
```
Comment (eric-ch) [inline src/policy.c:579]: SEGV here, rule can be `NULL`.

Triggers on [$10131](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10131) when plugging a USB thumbdrive when a VM is focused.
```
#0  policy_auto_assign_new_device (device=device@entry=0x428e80) at ../../git/src/policy.c:566
#1  0x0000000000406177 in udev_maybe_add_device (dev=dev@entry=0x42d420, auto_assign=auto_assign@entry=1) at ../../git/src/udev.c:494
#2  0x0000000000406775 in udev_event () at ../../git/src/udev.c:617
#3  0x0000000000403acf in main () at ../../git/src/main.c:135
p rule
$1 = (rule_t *) 0x0
```

`default_lookup` can fail and return `NULL`, maybe that's not intended?

**Review by eric-ch — CHANGES_REQUESTED** [review]

File: src/policy.c (hunk 1 above)
Comment (pearsonk) [reply src/policy.c:579]: Ah, it's not that default_lookup is failing, it's that the policy lookup after doesn't track the rule used for making the decision. Added a commit to add a parameter to policy_is_allowed to be able to keep track of the rule used to make the policy decision. Built and tested that it addresses the vm-focus auto-assign.

Comment (eric-ch) [reply src/policy.c:579]: Ah I misread that.

`policy_is_allowed` now has a more complex behavior:
- Allow rule: Returns `true` and set `*rules_ptr` if `rules_ptr` is not `null`;
- Deny rule: Returns `false` and set `*rules_ptr` if `rules_ptr` is not `null`;
- Default Deny catch-all: Returns `false` without touching `*rules_ptr`;

**Comment by eric-ch** [comment]
Build [$10132](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10132)

**Review by eric-ch — APPROVED** [review]
LGTM.

**Comment by eric-ch** [comment]
Will merge soon.

**Pull request merged**